File: server/model/vault.go
package model

import "time"

// Vault type constants.
const (
	VaultTypePassword       = "password"
	VaultTypePrivateKey     = "private_key"
	VaultTypeSSHCertificate = "ssh_certificate"
)

type Vault struct {
	ID          string    `json:"id"`
	Name        string    `json:"name"`
	Type        string    `json:"type"` // password | private_key | ssh_certificate
	Data        string    `json:"data"`
	Remark      string    `json:"remark"`
	Fingerprint string    `json:"fingerprint,omitempty"`
	RefCount    int       `json:"ref_count,omitempty"`
	CreatedAt   time.Time `json:"created_at"`
	UpdatedAt   time.Time `json:"updated_at"`
}

type Credential struct {
	Password   string `json:"password,omitempty"`
	PrivKey    string `json:"private_key,omitempty"`
	PublicKey  string `json:"public_key,omitempty"`
	Passphrase string `json:"passphrase,omitempty"`
	Cert       string `json:"certificate,omitempty"` // SSH certificate (OpenSSH format)
}

// VaultItem is the list/detail response payload without ciphertext.
type VaultItem struct {
	ID            string    `json:"id"`
	Name          string    `json:"name"`
	Type          string    `json:"type"`
	Username      string    `json:"username"`
	Remark        string    `json:"remark"`
	Fingerprint   string    `json:"fingerprint"`
	RefCount      int       `json:"ref_count"`
	HasPassphrase bool      `json:"has_passphrase"`
	CreatedAt     time.Time `json:"created_at"`
	UpdatedAt     time.Time `json:"updated_at"`
}

// VaultListFilter controls vault list query.
type VaultListFilter struct {
	Type string // empty = all
	Q    string // name/remark substring
}

// ProfileRef is a lightweight profile reference for vault deletion checks.
type ProfileRef struct {
	ID   string `json:"id"`
	Name string `json:"name"`
}

File: web/src/types/vault.ts
export type VaultType = 'password' | 'private_key'

export interface VaultItem {
  id: string
  name: string
  type: VaultType
  username: string
  remark: string
  fingerprint: string
  ref_count: number
  has_passphrase: boolean
  created_at: string
  updated_at: string
}

export interface VaultCredential {
  password?: string
  private_key?: string
  passphrase?: string
  public_key?: string
}

export interface VaultCreateRequest {
  name: string
  type: VaultType
  username?: string
  remark?: string
  password?: string
  private_key?: string
  public_key?: string
  passphrase?: string
}

export type VaultUpdateRequest = VaultCreateRequest

export interface GenerateKeyRequest {
  algo: 'rsa' | 'ed25519'
  bits?: number
  passphrase?: string
}

export interface GenerateKeyResponse {
  public_key: string
  private_key: string
  fingerprint: string
}

export interface ProfileRef {
  id: string
  name: string
}

export const VAULT_TYPE_LABELS: Record<VaultType, string> = {
  password: '密码',
  private_key: '私钥',
}

export const VAULT_TYPE_ALL = 'all' as const
export type VaultFilterType = VaultType | typeof VAULT_TYPE_ALL
